meta-refkit/lib/oeqa/selftest/secureboot.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+# ex:ts=4:sw=4:sts=4:et
+# -*- tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*-
+#
+# Copyright (c) 2017, Intel Corporation.
+# All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# AUTHORS
+# Mikko Ylinen <mikko.ylinen@linux.intel.com>
+#
+# Based on meta/lib/oeqa/selftest/* and meta-refkit/lib/oeqa/selftest/*
+
+"""Test cases for secure boot with QEMU running OVMF."""
+
+import os
+import unittest
+import re
+
+from oeqa.selftest.base import oeSelfTest
+from oeqa.utils.commands import runCmd, bitbake, get_bb_var, runqemu
+
+class SecureBootTests(oeSelfTest):
+    """Secure Boot test class."""
+
+    ovmf_keys_enrolled = False
+    test_image = 'refkit-image-common'
+
+    def setUpLocal(self):
+
+        if not SecureBootTests.ovmf_keys_enrolled:
+            print('\n')
+            print('Building ovmf-shell-image-enrollkeys')
+            bitbake('ovmf-shell-image-enrollkeys')
+            print('Building ovmf')
+            bitbake('ovmf')
+
+            cmd = ("runqemu "
+                   "ovmf-shell-image-enrollkeys wic intel-corei7-64 "
+                   "ovmf.secboot nographic serial slirp")
+            print('Running "%s"' % cmd)
+            status = runCmd(cmd)
+
+            # Debug
+            print("EFI shell dump:\n %s" % status.output)
+
+            if not re.search('info: success', status.output, re.M):
+                print("EFI shell dump:\n %s" % status.output)
+            else:
+                # keys enrolled in ovmf.secboot.vars
+                SecureBootTests.ovmf_keys_enrolled = True
+
+    @classmethod
+    def tearDownLocal(self):
+        # Seems this is mandatory between the tests (a signed image is booted
+        # when running test_boot_unsigned_image after test_boot_signed_image).
+        bitbake('-c clean %s' % self.test_image)
+
+    @classmethod
+    def tearDownClass(self):
+
+        print('Cleaning: remove ovmf-shell-image-enrollkeys')
+        bitbake('-c cleanall ovmf-shell-image-enrollkeys')
+        print('Cleaning: remove ovmf')
+        bitbake('-c clean ovmf')
+        print('Cleaning: remove ovmf SSTATE')
+        bitbake('-c cleansstate ovmf')
+
```

Comment:
I'll look into that once I get this working in CI. 

OVMF still puts EFI shell boot target *before* the hard drive so it never boots the image but enters the shell.